bot/utils/blurple_formatter.py

```diff
@@ -0,0 +1,717 @@
+    return " " * randrange(a, b)
+
+
+def num_spaces(src: str) -> int:
+    """Count the number of spaces at the beginning of the string."""
+    return len(src) - len(src.lstrip())
+
+
+def invert_indents(src: str) -> str:
+    """
+    Invert the indentation on each line of the given string.
+
+    Lines with the maximum indentation become least indented, and vice versa.
+    """
+    lines = src.splitlines()
+    max_spaces = max(num_spaces(line) for line in lines)
+    return "\n".join(
+        (max_spaces - num_spaces(line)) * " " + line.lstrip() for line in lines
+    )
+
+
+def parenthesize(node: ast.AST, _nl_able: bool = False) -> str:
+    """Wrap the un-parsed node in parentheses."""
+    return f"({unparse(node, True)})"
+
+
+def indent(nodes: list[ast.AST], n: int) -> Generator[str, None, None]:
+    """
+    Indent each line of the un-parsed nodes by the given number of spaces.
+
+    Multiple nodes are put on the same line, separated by semicolons, if possible
+    until they hit the line length limit.
+    """
+    curr_chunk = []
+    curr_len = 0
+    for stmt in nodes:
+        stmt_lines = unparse(stmt).splitlines()
+        if isinstance(stmt, semicolon_able):
+            for line in stmt_lines:
+                curr_chunk.append(line)
+                curr_len += len(line)
+                if curr_len > max_line_length or curr_chunk[-1][-1] != ";":
+                    yield f"{n * ' '}{f'{space()}'.join(curr_chunk)}"
+                    curr_len = 0
+                    curr_chunk = []
+        else:
+            yield f"{n * ' '}{f'{space()}'.join(curr_chunk)}"
+            yield from (f"{n * ' '}{line}" for line in stmt_lines)
+            curr_len = 0
+            curr_chunk = []
+    yield f"{n * ' '}{f'{space()}'.join(curr_chunk)}"
+
+
+def unparse(node: ast.AST, nl_able: bool = False) -> str:
+    """Convert the ast node to formatted(!) Python source code."""
+    if isinstance(node, ast.AnnAssign):
+        target, annotation, value = node.target, node.annotation, node.value
+
+        return (
+            f"{unparse(target)}{space()}:{space()}{unparse(annotation)}"
+            + (f"{space()}={space()}{unparse(value)}" if value else "")
+            + f"{space()};"
+        )
+    if isinstance(node, ast.Assert):
+        test, msg = node.test, node.msg
+
+        return (
+            f"assert{space(1)}{test}"
+            + (f",{space()}{unparse(msg)}" if msg else "")
+            + f"{space()};"
+        )
+    if isinstance(node, ast.Assign):
+        targets, value = node.targets, node.value
+
+        return (
+            f"{f'{space()}={space()}'.join(map(unparse, targets))}{space()}={space()}{unparse(value)}"
+            + f"{space()};"
+        )
+    if isinstance(node, ast.AsyncFor):
+        target, iter_, body, orelse = node.target, node.iter, node.body, node.orelse
+
+        s = [
+            f"async{space()}for{space()}{unparse(target)}{space()}in{space()}{unparse(iter_)}{space()}:",
```

Comment:
Depending on whether you change `space()` defaults per my last comment, these may require you adding `1` for some.